Building charm with cmake currently fails on Frontera

```

login4.frontera(1292)$ module list

Currently Loaded Modules:
  1) git/2.24.1      3) cmake/3.16.1   5) hwloc/1.11.12   7) TACC        9) boost/1.69  11) impi/19.0.5
  2) autotools/1.2   4) pmix/3.1.4     6) xalt/2.7.28     8) gcc/9.1.0  10) ucx/1.8.0   12) python3/3.7.0

```


```
login4.frontera(1293)$ cmake --version
cmake version 3.16.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).

```

```
login4.frontera(1291)$ ./build charm++ mpi-linux-x86_64 smp --enable-error-checking --suffix=impi-prod  --with-production  --basedir=$TACC_IMPI_DIR/intel64/ -j20 2>&1 | tee res
Build options for mpi-linux-x86_64:
  opt_CC =
  opt_CXX =
  opt_FC =
  opt_ampi_error_checking = 0
  opt_ampi_mpich_tests = 0
  opt_ampi_only = 0
  opt_build_shared = 0
  opt_ccs = 0
  opt_charmdebug = 0
  opt_charmpy = 0
  opt_compiler =
  opt_controlpoint = 0
  opt_cuda = 0
  opt_drone_mode = 0
  opt_error_checking = 1
  opt_extra_opts =
  opt_incdir = -I/opt/intel/compilers_and_libraries_2019.5.281/linux/mpi/intel64//include
  opt_lbtime_type = double
  opt_lbuserdata = 0
  opt_libdir = -L/opt/intel/compilers_and_libraries_2019.5.281/linux/mpi/intel64//lib
  opt_lockless = 0
  opt_mempool_cutoff = 26
  opt_network = mpi
  opt_numa = 0
  opt_omp = 0
  opt_parallel = -j20
  opt_pedantic = 0
  opt_prio_type = bitvec
  opt_production = Release
  opt_pthreads = 0
  opt_qlogic = 0
  opt_randomized_msgq = 0
  opt_refnum_type = unsigned short
  opt_replay = 0
  opt_romio = 1
  opt_shrinkexpand = 0
  opt_smp = 1
  opt_stats = 0
  opt_suffix = -impi-prod
  opt_syncft = 0
  opt_task_queue = 0
  opt_tcp = 0
  opt_tracing = 0
  opt_tracing_commthread = 0
  opt_ucx_pmi =
  opt_zlib = 1
-- The CXX compiler identification is GNU 9.1.0
-- The C compiler identification is GNU 9.1.0
-- Check for working CXX compiler: /opt/apps/gcc/9.1.0/bin/c++
-- Check for working CXX compiler: /opt/apps/gcc/9.1.0/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /opt/apps/gcc/9.1.0/bin/gcc
-- Check for working C compiler: /opt/apps/gcc/9.1.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Looking for a Fortran compiler
-- Looking for a Fortran compiler - /bin/f95
-- The Fortran compiler identification is GNU 4.8.5
-- Check for working Fortran compiler: /bin/f95
-- Check for working Fortran compiler: /bin/f95  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /bin/f95 supports Fortran 90
-- Checking whether /bin/f95 supports Fortran 90 -- yes
-- Detecting Fortran/C Interface
-- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
-- Verifying Fortran/C Compiler Compatibility
-- Verifying Fortran/C Compiler Compatibility - Success
-- Verifying Fortran/CXX Compiler Compatibility
-- Verifying Fortran/CXX Compiler Compatibility - Failed
CMake Error at /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface.cmake:383 (message):
  The Fortran compiler:

    /bin/f95

  and the CXX compiler:

    /opt/apps/gcc/9.1.0/bin/c++

  failed to compile a simple test project using both languages.  The output
  was:

    Change Dir: /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX

    Run Build Command(s):/bin/gmake VerifyFortranC && /opt/apps/cmake/3.16.1/bin/cmake -S/opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify -B/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX --check-build-system CMakeFiles/Makefile.cmake 0
    /bin/gmake -f CMakeFiles/Makefile2 VerifyFortranC
    gmake[1]: Entering directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    /opt/apps/cmake/3.16.1/bin/cmake -S/opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify -B/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX --check-build-system CMakeFiles/Makefile.cmake 0
    /opt/apps/cmake/3.16.1/bin/cmake -E cmake_progress_start /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX/CMakeFiles 6
    /bin/gmake -f CMakeFiles/Makefile2 CMakeFiles/VerifyFortranC.dir/all
    gmake[2]: Entering directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    /bin/gmake -f CMakeFiles/VerifyFortran.dir/build.make CMakeFiles/VerifyFortran.dir/depend
    gmake[3]: Entering directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    cd /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX && /opt/apps/cmake/3.16.1/bin/cmake -E cmake_depends "Unix Makefiles" /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX/CMakeFiles/VerifyFortran.dir/DependInfo.cmake
    Scanning dependencies of target VerifyFortran
    gmake[3]: Leaving directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    /bin/gmake -f CMakeFiles/VerifyFortran.dir/build.make CMakeFiles/VerifyFortran.dir/build
    gmake[3]: Entering directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    [ 16%] Building Fortran object CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    /bin/f95 -DVERIFY_CXX -I/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX  -O3 -DNDEBUG -O3   -c /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify/VerifyFortran.f -o CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    [ 33%] Linking Fortran static library libVerifyFortran.a
    /opt/apps/cmake/3.16.1/bin/cmake -P CMakeFiles/VerifyFortran.dir/cmake_clean_target.cmake
    /opt/apps/cmake/3.16.1/bin/cmake -E cmake_link_script CMakeFiles/VerifyFortran.dir/link.txt --verbose=1
    /opt/apps/gcc/9.1.0/bin/ar qc libVerifyFortran.a  CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    /opt/apps/gcc/9.1.0/bin/ranlib libVerifyFortran.a
    gmake[3]: Leaving directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    [ 33%] Built target VerifyFortran
    /bin/gmake -f CMakeFiles/VerifyFortranC.dir/build.make CMakeFiles/VerifyFortranC.dir/depend
    gmake[3]: Entering directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    cd /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX && /opt/apps/cmake/3.16.1/bin/cmake -E cmake_depends "Unix Makefiles" /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX /scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX/CMakeFiles/VerifyFortranC.dir/DependInfo.cmake
    Scanning dependencies of target VerifyFortranC
    gmake[3]: Leaving directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    /bin/gmake -f CMakeFiles/VerifyFortranC.dir/build.make CMakeFiles/VerifyFortranC.dir/build
    gmake[3]: Entering directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    [ 50%] Building C object CMakeFiles/VerifyFortranC.dir/main.c.o
    /opt/apps/gcc/9.1.0/bin/gcc -DVERIFY_CXX -I/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX  -O3 -DNDEBUG   -o CMakeFiles/VerifyFortranC.dir/main.c.o   -c /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify/main.c
    [ 66%] Building C object CMakeFiles/VerifyFortranC.dir/VerifyC.c.o
    /opt/apps/gcc/9.1.0/bin/gcc -DVERIFY_CXX -I/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX  -O3 -DNDEBUG   -o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o   -c /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify/VerifyC.c
    [ 83%] Building CXX object CMakeFiles/VerifyFortranC.dir/VerifyCXX.cxx.o
    /opt/apps/gcc/9.1.0/bin/c++  -DVERIFY_CXX -I/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX  -O3 -DNDEBUG   -o CMakeFiles/VerifyFortranC.dir/VerifyCXX.cxx.o -c /opt/apps/cmake/3.16.1/share/cmake-3.16/Modules/FortranCInterface/Verify/VerifyCXX.cxx
    [100%] Linking CXX executable VerifyFortranC
    /opt/apps/cmake/3.16.1/bin/cmake -E cmake_link_script CMakeFiles/VerifyFortranC.dir/link.txt --verbose=1
    /opt/apps/gcc/9.1.0/bin/c++  -O3 -DNDEBUG   CMakeFiles/VerifyFortranC.dir/main.c.o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o CMakeFiles/VerifyFortranC.dir/VerifyCXX.cxx.o  -o VerifyFortranC   -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5  libVerifyFortran.a -lgfortran -lquadmath
    /opt/apps/gcc/9.1.0/bin/ld: CMakeFiles/VerifyFortranC.dir/VerifyCXX.cxx.o: in function `VerifyCXX':
    VerifyCXX.cxx:(.text+0x1b): undefined reference to `operator delete(void*, unsigned long)'
    collect2: error: ld returned 1 exit status
    gmake[3]: *** [VerifyFortranC] Error 1
    gmake[3]: Leaving directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    gmake[2]: *** [CMakeFiles/VerifyFortranC.dir/all] Error 2
    gmake[2]: Leaving directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    gmake[1]: *** [CMakeFiles/VerifyFortranC.dir/rule] Error 2
    gmake[1]: Leaving directory `/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/FortranCInterface/VerifyCXX'
    gmake: *** [VerifyFortranC] Error 2


Call Stack (most recent call first):
  CMakeLists.txt:32 (FortranCInterface_VERIFY)


-- Configuring incomplete, errors occurred!
See also "/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/CMakeOutput.log".
See also "/scratch1/03808/nbhat4/charm/mpi-linux-x86_64-smp-impi-prod/CMakeFiles/CMakeError.log".
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,20 +18,17 @@ project(Charm++ LANGUAGES CXX C VERSION 6.10.1)
 find_package(Threads REQUIRED)
 find_package(OpenMP) # Do this before Fortran, in case we don't have a Fortran compiler
 
-
 # We need C++11 for (almost) all targets
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-include(CheckLanguage)
-check_language(Fortran)
+enable_language(Fortran OPTIONAL)
 if(CMAKE_Fortran_COMPILER)
-  enable_language(Fortran OPTIONAL)
   include(FortranCInterface)
   FortranCInterface_VERIFY()
   FortranCInterface_VERIFY(CXX)
 else()
-  message(STATUS "No Fortran support")
+  message(STATUS "Charm++: No Fortran support.")
 endif()
 
 if(${FortranCInterface_VERIFIED_CXX})
@@ -808,7 +805,7 @@ message("  C++ compiler:    ${CMAKE_CXX_COMPILER} [${CMAKE_CXX_COMPILER_ID} ${CM
 if(CMAKE_Fortran_COMPILER)
   message("  F90 compiler:    ${CMAKE_Fortran_COMPILER} [${CMAKE_Fortran_COMPILER_ID} ${CMAKE_Fortran_COMPILER_VERSION}]")
 else()
-  message("  F90 compiler:    N/A")
+  message("  F90 compiler:    ---")
 endif()
 message("  CMake:           ${CMAKE_COMMAND} [${CMAKE_VERSION}]")
 message("  Charmc flags:    ${MAIN_CFLAGS}")

